test-data/unit/check-deprecated.test

```diff
@@ -0,0 +1,353 @@
+-- Type checker test cases for reporting deprecations.
+
+[case testDeprecatedFunction]
+
+from typing_extensions import deprecated
+
+@deprecated("use f2 instead")
+def f() -> None: ...
+
+f  # E: __main__.f is deprecated - use f2 instead
+f(1)  # E: __main__.f is deprecated - use f2 instead \
+      # E: Too many arguments for "f"
+f[1]  # E: __main__.f is deprecated - use f2 instead \
+      # E: Value of type "Callable[[], None]" is not indexable
+g = f  # E: __main__.f is deprecated - use f2 instead
+g()  # E: __main__.f is deprecated - use f2 instead
```

Comment:
This ideally shouldn't error since we already got an error at the `g = f` line. I don't mind if that's too hard to implement though.

Reply:
Unfortunately, I have no spontaneous idea how to implement this with reasonable effort.

Reply:
@JelleZijlstra: Fixed by commit 49ee7a8ee53907b8b65665adcc756e811780811b.